pythran/analyses/pure_expressions.py

```diff
@@ -27,14 +27,14 @@ def visit_FunctionDef(self, node):
         return False
 
     def generic_visit(self, node):
-        is_pure = all(map(self.visit, ast.iter_child_nodes(node)))
+        is_pure = all([self.visit(x) for x in ast.iter_child_nodes(node)])
         if is_pure:
             self.result.add(node)
         return is_pure
 
     def visit_Call(self, node):
         # check if all arguments are Pures
-        is_pure = all(self.visit(arg) for arg in node.args)
+        is_pure = all([self.visit(arg) for arg in node.args])
```

Comment:
For a case like:
```python
foo(a, 2 + 4)
```
a is not pure but I want to keep going and looks other argument. Otherwise, `2 + 4` will not be marked pure and constant folding will not happen.